src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -488,12 +511,18 @@ export default class QueryStringInputUI extends Component<Props, State> {
       body: JSON.stringify({ opt_in: language === 'kuery' }),
     });
 
-    this.services.storage.set('kibana.userQueryLanguage', language);
+    const storageKey = this.props.storageKey;
```

Comment:
```suggestion
    this.services.storage.set(storageKey!, language);
```

We have a default prop so this can't be `undefined`